planet/src/main/java/com/google/planet/data/ItineraryGenerator.java

```diff
@@ -123,14 +117,56 @@ private int getTravelDurationInMinutes(Duration duration) {
                 items.add(item);
 
                 // Update the next event's start time
-                if (i != events.size() - 1) { 
-                    start += event.getDurationInMinutes() + getTravelDurationInMinutes(travelTimeGraph[i][i+1]);
-                }
+                start += event.getDurationInMinutes() + getTravelDurationInMinutes(directionsLegs[i].duration);
             }else {
                 errorMessage = "Sorry, you have too many events in a day!";
                 return Arrays.asList();
             }
         }
         return items;
     }
+
+    // Function that makes a request to the Directions API and return a direction route
+    // A DirectionsRoute has field legs, which is an array of DirectionsLegs
+    // Each DirectionsLeg contains starting address, ending address, and duration
+    // DirectionRoute also has int[] waypointOrder, which will be empty if optimized is set to false,
+    // and will contain the new order of waypoints if optimized is set to true
+    // For the MVP, real time traffic is NOT used.
+    private DirectionsRoute getDirectionsRoute(List<Event> events, boolean optimized) {
```

Comment:
This is an example where it is appropriate to have the optimized setting in the function parameter, since 99% of the function is identical regardless of the value here. 

However, we're still leaning heavily on primitives here. Consider making an ItineraryOrder enum and passing a value in here. That way, the call sites are more readable.

Imagine you're reading the code that calls this, and you have not read this function yet. Which call would be more readable?
`getDirectionsRoute(events, true)`
`getDirectionsRoute(events, ItineraryOrder.Optimized)` ?

Reply:
Okay!